### PR TITLE
Adds examples using the @first and @last Handlebars variables

### PR DESCRIPTION
**Pull request by davidwrede**
Addresses #3805.

Signed-off-by: David Wrede <dwrede@chef.io>

### CODE REVIEW

File: www/source/partials/docs/_built-in-helpers.html.md.erb [hunk 1]
```diff
@@ -79,4 +79,26 @@ like this:
...
     [[servers]]
     host = "host-2"
-    port = 3434
\ No newline at end of file
+    port = 3434
+
+And for both `each` and `unless`, you can use `@first` and `@last` to specify which item in an array you want to perform business logic on. For example:
+
+    "mongo": {
+      {{~#each bind.database.members as |member|}}
```
Comment (elliott-davis) [inline www/source/partials/docs/_built-in-helpers.html.md.erb]: I think in core plans were trying to standardize around `~` at the end of the handlebars syntax. https://github.com/habitat-sh/core-plans/pull/986/files

**Comment by davidwrede** [comment]
Feedback incorporated and ready for another review.

**Comment by elliott-davis** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/308559008).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/308559008).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**